Plots/MapProjections/NCL_sat_1.py

```diff
@@ -0,0 +1,380 @@
+"""
+NCL_sat_1.py
+===============
+This script illustrates the following concepts:
+   - Creating an orthographic projection
+   - Drawing line contours over a satellite map
+   - Manually labeling contours
+   - Transforming coordinates
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/sat_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/sat_1_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import xarray as xr
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import numpy as np
+from sklearn.cluster import DBSCAN
+import warnings
+
+import geocat.datafiles as gdf
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and
+# load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/slp.1963.nc"), decode_times=False)
+
+# Get data from the 24th timestep
+pressure = ds.slp[24, :, :]
+
+# Translate short values to float values
+pressure = pressure.astype('float64')
+
+# Convert Pa to hPa data
+pressure = pressure*0.01
+
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_pressure = gvutil.xr_add_cyclic_longitudes(pressure, "lon")
+
+###############################################################################
+
+
+def findLocalExtrema(da, highVal=1040, lowVal=975, eType='Low'):
+    """
+    Utility function to find local low/high field variable coordinates on a contour map. To classify as a local high, the data
+    point must be greater than highVal, and to classify as a local low, the data point must be less than lowVal.
+    Args:
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable (ex. pressure) data values
+        highVal (:class:`int`):
+            Data value that the local high must be greater than to qualify as a "local high" location.
+            Default highVal is a pressure value of 1040 hectopascals.
```

Comment:
Should we be including units here? Does the implementation make use of them in any way? For example, if the data units were in pascals would they be converted to hectopascals for comparison against highVal?

Reply:
It was suggested to me that I include them just so people would understand the default values, but I will take them out.